python/benchmarks/parse_bdd100k.py

```diff
@@ -0,0 +1,116 @@
+
+import os
+from pathlib import Path
+from typing import Union
+
+import click
+import lance
+import pandas as pd
+import pyarrow as pa
+import pyarrow.fs
+
+from bench_utils import DatasetConverter
+
+
+class BDD100kConverter(DatasetConverter):
+    def __init__(self, uri_root: Union[str, Path]):
+        super().__init__("bdd100k", uri_root)
+
+    def read_metadata(self) -> pd.DataFrame:
+        frames = []
+        for split in ["train", "val"]:
+            annotation = pd.read_json(
+                os.path.join(
+                    self.uri_root, "bdd100k", "labels", f"bdd100k_labels_images_{split}.json"
+                )
+            )
+            annotation["split"] = split
+            annotation["image_uri"] = annotation["name"].map(
+                lambda name: os.path.join(self.uri_root, "bdd100k", "images", "100k", split, name))
+            frames.append(annotation)
+
+        return pd.concat(frames)
+
+    def image_uris(self, table):
+        return table["image_uri"].to_numpy()
+
+    def get_schema(self):
+        schema = pa.schema(
+            [
+                ("name", pa.utf8()),
+                ("image_uri", pa.utf8()),
+                ("timestamp", pa.int32()),
+                ("split", pa.dictionary(pa.uint8(), pa.utf8())),
+                (
+                    "attributes",
+                    pa.struct(
+                        [
+                            ("weather", pa.dictionary(pa.uint8(), pa.utf8())),
+                            ("scene", pa.dictionary(pa.uint8(), pa.utf8())),
+                            ("timeofday", pa.dictionary(pa.uint8(), pa.utf8())),
+                        ]
+                    ),
+                ),
+                (
+                    "labels",
+                    pa.list_(
+                        pa.struct(
+                            [
+                                ("category", pa.dictionary(pa.int8(), pa.utf8())),
+                                (
+                                    "attributes",
+                                    pa.struct(
+                                        [
+                                            ("occluded", pa.bool_()),
+                                            ("truncated", pa.bool_()),
+                                            (
+                                                "trafficLightColor",
+                                                pa.dictionary(pa.uint8(), pa.utf8()),
+                                            ),
+                                        ]
+                                    ),
+                                ),
+                                ("manualShape", pa.bool_()),
+                                ("manualAttributes", pa.bool_()),
+                                ("box", pa.list_(pa.float32())),
+                            ]
+                        )
+                    ),
+                ),
+            ]
+        )
+        return schema
+
+
+@click.command
```

Comment:
should we make this a generic superclass method as well (like with benchmark suite)? Feels very similar between the 3 different benchmark cases.